src/metabase/xrays/automagic_dashboards/populate.clj

```diff
@@ -27,25 +26,17 @@
   "Default card height"
   6)
 
-(defn create-collection!
-  "Create and return a new collection."
-  [title description parent-collection-id]
-  (first (t2/insert-returning-instances!
-          'Collection
-          (merge
-           {:name        title
-            :description description}
-           (when parent-collection-id
-             {:location (collection/children-location (t2/select-one ['Collection :location :id]
-                                                                     :id parent-collection-id))})))))
-
-(defn get-or-create-root-container-collection
-  "Get or create container collection for automagic dashboards in the root collection."
-  []
-  (or (t2/select-one 'Collection
-                     :name     "Automatically Generated Dashboards"
-                     :location "/")
-      (create-collection! "Automatically Generated Dashboards" nil nil)))
+(defn get-or-create-container-collection
+  "Get or create container collection for automagic dashboards in a given location."
+  [location]
+  (or (t2/select-one :model/Collection
+                     :name "Automatically Generated Dashboards"
+                     :archived false
+                     :location location)
+      (t2/insert-returning-instance!
+       :model/Collection
+       {:name "Automatically Generated Dashboards"
+        :location location})))
```

Comment:
Is it ok that this will create an new one if the original container collection at location gets archived? If so, would it be better to update it?

``` clojure
[((juxt :id :location :name) (get-or-create-container-collection "/"))
 (t2/update! :model/Collection 8 {:archived true})
 ((juxt :id :location :name) (get-or-create-container-collection "/"))]
 
;;=>

[[8 "/" "Automatically Generated Dashboards"] 
 1 
 [9 "/" "Automatically Generated Dashboards"]]
```

Reply:
Maybe I should add a comment here explaining this. I think it's better to just create the new one rather than unarchiving the old one, because:
- if you unarchive the old collection in a nonstandard way, the contents will still be archived, and then will be in an awkward state (not visible in the Trash because their collection isn't there, not visible outside the Trash because they're archived)
- if you archive the old collection in the standard way, the contents that were archived with it will be unarchived, which would be annoying if you'd stuck the whole thing in the trash because you didn't use it

Reply:
Makes sense